hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/OzoneConfigKeys.java

```diff
@@ -123,6 +123,12 @@
    * */
   public static final String OZONE_ADMINISTRATORS_WILDCARD = "*";
 
+  public static final String OZONE_CLIENT_STREAM_BUFFER_SIZE =
```

Comment:
Any specific reason to define a new config rather than using the chunk config itself?

Reply:
Ideally the file-per-chunk layout could be improved to use the existing chunk size config, eg. even if client sends 64KB chunks, server writes 4MB chunk files.

Reply:
> Ideally the file-per-chunk layout could be improved to use the existing chunk size config, eg. even if client sends 64KB chunks, server writes 4MB chunk files.

I like the idea of a new config , but with different config of chunk size and buffer size can lead to a case where ratis client itself can not send any request to server if buffer size exceeds chunk size. We can enforce client buffer size <= chunk size in the code itself.

Reply:
> We can enforce client buffer size <= chunk size in the code itself.

I'll update the checks in `BlockOutputStreamEntryPool` with this.  Does this need to consider actual `ozone.scm.chunk.size` config or the max. chunk size (32MB) allowed, which is used for setting max. GRPC message size?

Reply:
I think its ok to restrict the buffer size check to chunk size itself. Currently it is set to MAX_CHUNK_SIZE = 32 MB, but it can also be changed in DatanodeGrpcConfig.